packages/app-config-writer/src/preview-config/prerequisites.ts

```diff
@@ -4,6 +4,29 @@ import type { Editor } from 'mem-fs-editor';
 import type { Package } from '@sap-ux/project-access';
 import type { PromptObject } from 'prompts';
 import type { ToolsLogger } from '@sap-ux/logger';
+import { satisfies } from 'semver';
+
+/**
+ * Check if the version of the given package is lower than the minimal version.
+ *
+ * @param packageJson - the package.json file content
+ * @param dependencyName - the name of the (dev)dependency to check
+ * @param minVersionInfo - the minimal version to check against
+ * @param mandatory - (default true) if the existence of the dependency is mandatory
+ * @returns indicator if the version is lower than the minimal version
+ */
+function isLowerThanMinimalVersion(
+    packageJson: Package,
+    dependencyName: string,
+    minVersionInfo: string,
+    mandatory: boolean = true
+): boolean {
+    const versionInfo = packageJson?.devDependencies?.[dependencyName] ?? packageJson?.dependencies?.[dependencyName];
+    if (!versionInfo) {
+        return mandatory;
```

Comment:
Can you explain this from a consumer perspective? If I `mandatory` is `true` and the module is not found in dependencies of devDependencies, function `isLowerThanMinimalVersion()` returns true?

Reply:
The question was what to do in case a (dev)dependency is mandatory but has not been found. I thought it might be best to behave as if the version was too low. This means the user will get [an error message](https://github.com/SAP/open-ux-tools/blob/3888918983c348ee8ab8aa5bc583d3e28b1bf8e7/packages/app-config-writer/src/preview-config/prerequisites.ts#L63) with the name of the missing package and the lowest version needed.

e.g. `UI5 CLI version 3.0.0 or higher is required to convert the preview to virtual files. For more information, see https://sap.github.io/ui5-tooling/v3/updates/migrate-v3.`

An alternative would be to throw an exception. But as we would most likely use exactly the same error message I didn't see a real benefit. Would you prefer the exception?